html/modules/custom/reliefweb_import/src/Command/ReliefwebImportCommand.php

```diff
@@ -0,0 +1,581 @@
+        $this->logger()->info('Nothing to do for @name.', [
+          '@name' => $label(),
+        ]);
+
+        return;
+      }
+
+      return $body;
+    }
+    catch (ClientException $exception) {
+      $this->logger()->error('Unable to process @name, got @code: @message.', [
+        '@name' => $label(),
+        '@code' => $exception->getCode(),
+        '@message' => $exception->getMessage(),
+      ]);
+    }
+    catch (\Exception $exception) {
+      $this->logger()->error('Unable to process @name, got @code: @message.', [
+        '@name' => $label(),
+        '@code' => $exception->getCode(),
+        '@message' => $exception->getMessage(),
+      ]);
+    }
+  }
+
+  /**
+   * Process XML data.
+   */
+  protected function processXml($body) {
+    $xml = new \SimpleXMLElement($body);
+    foreach ($xml as $item) {
+      // Check if job already exist.
+      if ($this->jobExists((string) $item->link)) {
+        $this->logger()->notice('Update existing job');
+        $this->updateJob($this->loadJobById((string) $item->link), $item);
+      }
+      else {
+        $this->logger()->notice('Create new job');
+        $this->createJob($item);
+      }
+    }
+  }
+
+  /**
+   * Check if job exists.
+   */
+  protected function jobExists($id) {
+    $sql = "SELECT COUNT(entity_id)
+      FROM {node__field_job_id}
+      WHERE bundle = 'job'
+      AND field_job_id_value = :id";
+
+    $count = $this->database->query($sql, [
+      ':id' => $id,
+    ])->fetchCol();
+
+    return $count[0] != 0;
+  }
+
+  /**
+   * Load job by Id.
+   */
+  protected function loadJobById($id) {
+    $sql = "SELECT entity_id
+      FROM {node__field_job_id}
+      WHERE bundle = 'job'
+      AND field_job_id_value = :id";
+
+    $nid = $this->database->query($sql, [
+      ':id' => $id,
+    ])->fetchCol();
+
+    return Job::load(reset($nid));
+  }
+
+  /**
+   * Create a new job.
+   */
+  protected function createJob($data) {
+    $values = [
+      'type' => 'job',
+      'uid' => 1,
+      'field_job_id' => (string) $data->link,
+      'title' => (string) $data->title,
+      'field_career_categories' => $data->field_career_categories[0] ? (array) $data->field_career_categories : 0,
+      'field_city' => (string) $data->field_city,
+      'field_job_closing_date' => (string) $data->field_job_closing_date,
+      'field_country' => $this->mapCountries((array) $data->field_country),
+      'field_how_to_apply' => (string) $data->field_how_to_apply,
+      'body' => [
+        'value' => $this->validateBody((string) $data->body),
+        'format' => 'markdown',
+      ],
+      'field_job_type' => $data->field_job_type[0] ? (array) $data->field_job_type : 0,
+      'field_job_experience' => $data->field_job_experience[0] ? (array) $data->field_job_experience : 0,
+      'field_source' => $data->field_source[0] ? (array) $data->field_source : 0,
+      'field_theme' => $data->field_theme[0] ? (array) $data->field_theme : 0,
```

Comment:
Nitpicking but casting to `array` seems unnecessary here and above. If `$data->field_theme[0]`(for example) is set then `$data->field_theme` is an array.

Also, shouldn't the default be `[]` rather than `0`?

Reply:
Ok, thanks.